src/tests/Microsoft.Diagnostics.Monitoring.EventPipe/PipelineTestUtilities.cs

```diff
@@ -10,25 +10,29 @@ namespace Microsoft.Diagnostics.Monitoring.EventPipe.UnitTests
 {
     internal static class PipelineTestUtilities
     {
-        public static async Task ExecutePipelineWithDebugee(Pipeline pipeline, RemoteTestExecution testExecution)
+        public static async Task ExecutePipelineWithDebugee(Pipeline pipeline, RemoteTestExecution testExecution, CancellationToken token = default)
         {
-            Task processingTask = pipeline.RunAsync(CancellationToken.None);
+            Task processingTask = pipeline.RunAsync(token);
 
             //Begin event production
             testExecution.SendSignal();
 
             //Wait for event production to be done
             testExecution.WaitForSignal();
 
-            //Signal for the pipeline to stop
-            await pipeline.StopAsync();
-
-            //After a pipeline is stopped, we should expect the RunTask to eventually finish
-            await processingTask;
-
-            //Signal for debugee that's ok to end/move on.
-            testExecution.SendSignal();
+            try
+            {
+                //Signal for the pipeline to stop
+                await pipeline.StopAsync(token);
 
+                //After a pipeline is stopped, we should expect the RunTask to eventually finish
+                await processingTask;
+            }
+            finally
+            {
+                //Signal for debugee that's ok to end/move on.
+                testExecution.SendSignal();
```

Comment:
Looks like the test failures are happening here. If the remote process dies or exits before this is hit for any reason it would cause the stack we're seeing.  It seems to only be happening on non-Windows platforms.

Reply:
Yep, I'm trying to get additional data about the target process. I'm not sure why it would die early in this scenario.